### PR TITLE
Added removal of both temporary file and directory which are created …

### PR DESCRIPTION
**Pull request by SieGeL2k16**
…automatically during upload of media data via API

### 1. Why is this change necessary?

If media data is uploaded via API Shopware creates first a tempory file, removes this file immediatly and creates a temporary directory under the same name. The media is then copied into this directory for further processing. However after media data is added to Shopware, these temporary directories are not removed afterwards! This is a massive problem if you bulk-add a lot of articles with images, as the temporary directory gets filled quickly with orphaned files. One of our servers had more than 350.000 temporary directories which occupied nearly the complete hd storage!

### 2. What does this change do, exactly?

It first unlink() the temporary file and afterwards rmdir() the created temporary directory.

### 3. Describe each step to reproduce the issue or behaviour.

Add an article via API with at least one image included and watch your temporary directory. You can see directories being created for every image. But these directories won't disappear after the article was uploaded.

### 4. Please link to the relevant issues (if any).

No issue opened (yet), we do require a fast solution to this problem.

### 5. Which documentation changes (if any) need to be made because of this PR?

No documentation changes required.

### 6. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Shopware/Components/Api/Resource/Media.php [hunk 1]
```diff
@@ -277,6 +281,8 @@ public function internalCreateMediaByFileLink($link, $albumId = -1)
             $manager->createMediaThumbnail($media, [], true);
         }
 
+        unlink($file);
+        rmdir(dirname($file));
```
Comment (oktupol) [inline engine/Shopware/Components/Api/Resource/Media.php]: You could combine the latter two changes in a `finally` block to avoid duplicate code

Comment (SieGeL2k16) [reply engine/Shopware/Components/Api/Resource/Media.php]: Thanks for the hint, I've added the finally block.

**Comment by SieGeL2k16** [comment]
sorry, looks like I've forgot to pull first the latest changes :( I'm a bit lost here, what should i do to fix it finally?

**Comment by soebbing** [comment]
I think the best way to fix this would be to copy the Sha of the relevant commit, delete the branch, create a new branch from 5.5 and cherry-pick the Sha of your commit. Then do a force push on the branch of the pr.

**Comment by SieGeL2k16** [comment]
@soebbing I've re-added the pull-request, hopefully now everything is fine.

**Pull request closed without merging**